iothub/device/src/Authentication/Security/SharedAccessSignatureBuilder.cs

```diff
@@ -145,5 +146,38 @@ internal static string Sign(string requestString, string key)
             using var algorithm = new HMACSHA256(Convert.FromBase64String(key));
             return Convert.ToBase64String(algorithm.ComputeHash(Encoding.UTF8.GetBytes(requestString)));
         }
+
+        internal static string GetSignature(string encodedURI, string key, long expiry)
+        {
+            return Sign($"{encodedURI}\n{expiry}", key);
+        }
+
+        internal static string GetToken(string encodedURI, string key, long expiry = 0)
+        {
+            long expiryValue = (expiry == 0) ? DateTimeOffset.UtcNow.AddMinutes(DEFAULT_TOKEN_EXPIRY_MINS).ToUnixTimeSeconds() : expiry;
+            string sig = WebUtility.UrlEncode(Sign($"{encodedURI}\n{expiryValue}", key));
+
+            return $"SharedAccessSignature sr={encodedURI}&sig={sig}&se={expiryValue}";
+        }
+
+        internal static string GetDeviceToken(string hostname, string deviceId, string key, long expiry = 0)
+        {
+            return GetToken(GetDeviceResourceURI(hostname, deviceId), key, expiry);
+        }
+
+        internal static string GetDeviceToken(string hostname, string deviceId, string moduleId, string key, long expiry = 0)
+        {
+            return GetToken(GetDeviceResourceURI(hostname, deviceId, moduleId), key, expiry);
+        }
```

Comment:
Instead of creating additional overloads for each of these methods you can use default values for params and have a single implementation.